storm-core/src/jvm/backtype/storm/scheduler/resource/RAS_Node.java

```diff
@@ -230,6 +241,61 @@ public void freeTopology(String topId, Cluster cluster) {
         _topIdToUsedSlots.remove(topId);
     }
 
+    public void freeMemory(double amount) {
+        _availMemory += amount;
+        LOG.debug("freeing {} memory on node {}...avail mem: {}", amount, this.getHostname(), _availMemory);
+        if (_availMemory > this.getTotalMemoryResources()) {
+            LOG.warn("Freeing more memory than there exists!");
+        }
+    }
+
+    public void freeCPU(double amount) {
+        _availCPU += amount;
+        LOG.debug("freeing {} CPU on node...avail CPU: {}", amount, this.getHostname(), _availCPU);
+        if (_availCPU > this.getAvailableCpuResources()) {
+            LOG.warn("Freeing more memory than there exists!");
+        }
+    }
```

Comment:
Why are freeMemory and freeCPU public?  They seem like they should be private or package private.